src/NBitcoin/BitcoinCore/Coins.cs

```diff
@@ -160,7 +160,7 @@ public void ReadWrite(BitcoinStream stream)
                 // coinbase height
                 stream.ReadWriteAsVarInt(ref this.nHeight);
 
-                if (stream.ConsensusFactory.Consensus.IsProofOfStake)
+                if (stream.ConsensusFactory is PosConsensusFactory)
```

Comment:
The assumption here would be that if the network's consensus factory is POS then it is a POS network? @dangershony this change won't break anything (currently) but do you agree with this? What if we on a POS network create a Pow consensus factory? I'm just trying to think how someone can mess this up accidentally.